src/app/services/actions/permissions-action-creator.ts

```diff
@@ -9,47 +13,60 @@ export function fetchScopesSuccess(response: object): IAction {
   };
 }
 
+export function fetchScopesPending(): any {
+  return {
+    type: FETCH_SCOPES_PENDING,
+  };
+}
+
 export function fetchScopesError(response: object): IAction {
   return {
     type: FETCH_SCOPES_ERROR,
     response,
   };
 }
 
-export function fetchScopes(): Function {
+export function fetchScopes(query?: IQuery): Function {
   return async (dispatch: Function, getState: Function) => {
-    const { sampleQuery: { sampleUrl, selectedVerb } } = getState();
-    const urlObject: URL = new URL(sampleUrl);
-    const createdAt = new Date().toISOString();
-    // remove the prefix i.e. beta or v1.0 and any possible extra '/' character at the end
-    const requestUrl = urlObject.pathname.substr(5).replace(/\/$/, '');
-    const permissionsUrl = 'https://graphexplorerapi.azurewebsites.net/api/GraphExplorerPermissions?requesturl=' +
-      requestUrl + '&method=' + selectedVerb;
-
-    const headers = {
-      'Content-Type': 'application/json',
-    };
-
-    const options: IRequestOptions = { headers };
-
-    return fetch(permissionsUrl, options)
-      .then(res => res.json())
-      .then(res => {
-        if (res.error) {
-          throw (res.error);
+    try {
+      const { devxApi } = getState();
+      let permissionsUrl = `${devxApi}/permissions`;
+
+      if (query) {
+        const { requestUrl, sampleUrl } = parseSampleUrl(query.sampleUrl);
+
+        if (!sampleUrl) {
+          throw new Error('url is invalid');
         }
-        dispatch(fetchScopesSuccess(res));
-      })
-      .catch(() => {
-        const duration = (new Date()).getTime() - new Date(createdAt).getTime();
-        const response = {
-          /* Return 'Forbidden' regardless of error, as this was a
-           permission-centric operation with regards to user context */
-          statusText: 'Forbidden',
-          status: '403',
-          duration
-        };
-        return dispatch(fetchScopesError(response));
-      });
+
+        permissionsUrl = `${permissionsUrl}?requesturl=/${requestUrl}&method=${query.selectedVerb}`;
+      }
+
+      const headers = {
+        'Content-Type': 'application/json',
+      };
+      const options: IRequestOptions = { headers };
+
+      dispatch(fetchScopesPending());
+
+      const response = await fetch(permissionsUrl, options);
+      if (response.ok) {
+        const scopes = await response.json();
+        return dispatch(fetchScopesSuccess(scopes));
+      }
+      throw (response);
+    } catch (error) {
+      return dispatch(fetchScopesError(error));
+    }
+  };
+}
+
+export function consentToScopes(scopes: string[]): Function {
+  return async (dispatch: Function) => {
+      const authResponse = await acquireNewAccessToken(scopes);
+      if (authResponse && authResponse.accessToken) {
```

Comment:
Is it possible to have an `authResponse` that doesn't have an `accessToken`?

Reply:
@jobala I want to say there is :-)

Reply:
Lets not repeat ourselves